README.md

```diff
@@ -1,3 +1,244 @@
-# Backend app
+# Backend app - Music Library
 
-This app was created with `@command-shift/create-backend-app`
+This app was created with `@command-shift/create-backend-app`. It is a music library API created with the CRUD REST API operations using Node.js, Express, Mocha-Chai, pgAdmin, and Postman.
+
+## Concepts of the app
+
+- Database migration and design
+- PostgreSQL
+- Creating and building databases in Docker container
+- Creating APIs with CRUD operations on databases
+- Integration testing with Mocha, Chai, and supertest
+- Using Postman to manage API requests
+- Nodemon to automatically restart app whenever changes are made to code
+- Using .dotenv to secure sensitive information
+
+## Getting started
+
+Fork and clone the repo, then run the following commands. Use pgAdmin and Postman to check whether CRUD operations and REST APIs are working
+
+```bash
+npm test # to test the code
+npm start # to start the app at http://localhost:3000
+```
+
+## API Endpoints
+
+### Artist
+
+<details>
+<summary><code>POST</code> <code><b>/artists</b></code> <code>add a new artist</code></summary>
+
+#### Parameters and body content
+
+> | Parameters | Body                          |
+> | ---------- | ----------------------------- |
+> | None       | name [string], genre [string] |
+
+#### Responses
+
+> | code  | description          |
+> | ----- | -------------------- |
+> | `201` | successful operation |
+
+</details>
+
+<details>
+<summary><code>GET</code> <code><b>/artists</b></code> <code>find all artists</code></summary>
+
+#### Parameters and body
+
+> None
+
+#### Responses
+
+> | code  | description          |
+> | ----- | -------------------- |
+> | `200` | successful operation |
+
+</details>
+
+<details>
+<summary><code>GET</code> <code><b>/artists/{id}</b></code> <code>find an artist by ID</code></summary>
+</details>
```

Comment:
This part isn't displaying quite right. I think this needs moving to about line 64